docs/operations/clean-metadata-store.md

```diff
@@ -0,0 +1,168 @@
+ - `druid.coordinator.kill.compaction.period`: Defines the frequency in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) for the cleanup job to check for and delete eligible compaction configuration records. Defaults to `P1D`.
+
+>If you already have an extremely large compaction configuration, you may not be able to delete compaction configuration due to size limits with the audit log. In this case you can set `druid.audit.manager.maxPayloadSizeBytes` and `druid.audit.manager.skipNullField` to avoid the auditing issue. See [Audit logging](../configuration/index.md#audit-logging).
+
+### Datasource records created by supervisors
+Datasource records created by supervisors become eligible for deletion when the supervisor is terminated or does not exist in the `druid_supervisors` table and the `durationToRetain` time has passed since their creation.
+
+Datasource cleanup uses the following configuration:
+ - `druid.coordinator.kill.datasource.on`: When `true`, enables cleanup datasources created by supervisors.
+ - `druid.coordinator.kill.datasource.period`: Defines the frequency in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) for the cleanup job to check for and delete eligible datasource records. Defaults to `P1D`.
+ - `druid.coordinator.kill.datasource.durationToRetain`: Defines the retention period in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) after creation that datasource records become eligible for deletion.
+
+### Indexer task logs
+The Druid Overlord handles task log metadata management.
+
+Indexer task log cleanup on the Overlord uses the following configuration:
+- `druid.indexer.logs.kill.enabled`: When `true`, enables cleanup of task logs.
+- `druid.indexer.logs.kill.durationToRetain`: Defines the length of time in milliseconds to retain task logs.
+- `druid.indexer.logs.kill.initialDelay`: Defines the length of time in milliseconds after the Overlord starts before it executes its first job to kill task logs.
+- `druid.indexer.logs.kill.delay`: The length of time in milliseconds between jobs to kill task logs.
+
+For more detail, see [Task logging](../configuration/index.md#task-logging).
+
+## Automated metadata cleanup example configuration
+Consider a scenario where you have scripts to create and delete hundreds of datasources and related entities a day. You do not want to fill your metadata store with leftover records. The datasources and related entities tend to persist for only one or two days. Therefore, you want to run a cleanup job that identifies and removes leftover records that are at least four days old. The exception is for audit logs, which you need to retain for 30 days:
+
+```
+...
+# Schedule the metadata management store task for every hour:
+druid.coordinator.period.metadataStoreManagementPeriod=P1H
+
+# Set a kill task to poll every day to delete Segment records
```

Comment:
For completeness. When `druid.coordinator.kill.on` is set to true, then either `killAllDataSources` or `killDataSourceWhitelist` in the dynamic configuration must be set. For this example, we can say that `killAllDataSources` is set to true.